cst438_register/src/test/java/com/cst438/JunitTestStudent.java

```diff
@@ -0,0 +1,154 @@
+package com.cst438;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.util.Optional;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import com.cst438.controller.StudentController;
+import com.cst438.domain.Student;
+import com.cst438.domain.StudentRepository;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+
+
+@ContextConfiguration(classes= {StudentController.class})
+@AutoConfigureMockMvc(addFilters=false)
+@WebMvcTest
+class JunitTestStudent {
+	
+	static final String URL = "http://localhost:8080";
+	public static final int TEST_STUDENT_ID = 1;
+	public static final String TEST_STUDENT_EMAIL = "test@csumb.edu";
+	public static final String TEST_STUDENT_NAME = "test";
+	
+	@MockBean
+	StudentRepository studentRepository;
+	
+	@Autowired
+	private MockMvc mvc;
+	
+
+	@Test
+	public void addStudent() throws Exception {
+		MockHttpServletResponse response;
+		
+		//we are adding to the database if not seen before so this should return null
+		given(studentRepository.findByEmail(TEST_STUDENT_EMAIL)).willReturn(null);
+		
+		response = mvc.perform(MockMvcRequestBuilders
+				.post("/student/?email="+TEST_STUDENT_EMAIL+"&name="+TEST_STUDENT_NAME)
+				.accept(MediaType.APPLICATION_JSON)).andReturn().getResponse();
+		
+		assertEquals(200, response.getStatus());
+		
+		Student result = fromJsonString(response.getContentAsString(),Student.class);
+		boolean found = false;
+		if(TEST_STUDENT_EMAIL.equals(result.getEmail())) {
+			found = true;
+		}
+		
+		assertEquals(true, found); //make sure we found the student by comparing their email addresses
+		
+		verify(studentRepository).save(any(Student.class)); //verify we called the save method
+		verify(studentRepository, times(1)).findByEmail(TEST_STUDENT_EMAIL); //verify the findbyemail was called
+	}
+	
+	@Test
+	public void addStudentAlreadyExists() throws Exception{
+		MockHttpServletResponse response;
+		Student student = new Student();
+		student.setName(TEST_STUDENT_NAME);
+		student.setEmail(TEST_STUDENT_EMAIL);
+		student.setStudent_id(TEST_STUDENT_ID);
```

Comment:
Student ID is initialized but never used in test case, consider removing variable from test

Reply:
it might be good to keep in case we need to lookup by student id instead of by student email since I believe only the id is guaranteed to be unique.